### PR TITLE
Upgrade Anchor

### PR DESCRIPTION
**Pull request by LucasSte**
This PR upgrades the Anchor version to v0.27.0. The IDL does not have the `state` field anymore.

The container CI image needs to be updated in https://github.com/hyperledger/solang-llvm/pull/9.

### CODE REVIEW

File: src/bin/idl/mod.rs [hunk 1]
```diff
@@ -310,7 +297,7 @@ fn instruction(
             )?;
         }
 
-        write!(f, ") {}external", if state { "" } else { "view " })?;
+        write!(f, ") external")?;
```
Comment (seanyoung) [inline src/bin/idl/mod.rs:300]: The ts code makes functions view *if* there are return values and there is no mutable account needed. We could do the same.

Comment (LucasSte) [reply src/bin/idl/mod.rs:300]: Good point. I'm going to add code to handle this.

File: src/bin/idl/mod.rs [hunk 2]
```diff
@@ -332,6 +320,18 @@ fn instruction(
...
+fn mutable_account_exists(accounts: &[IdlAccountItem]) -> bool {
+    let mut exists = false;
+    for item in accounts {
+        exists |= match item {
+            IdlAccountItem::IdlAccount(acc) => acc.is_mut,
+            IdlAccountItem::IdlAccounts(accs) => mutable_account_exists(&accs.accounts),
+        }
+    }
+
+    exists
```
Comment (seanyoung) [inline src/bin/idl/mod.rs:332]: Do you not like iterators?
```suggestion
    accounts.iter().any(|item| match item {
            IdlAccountItem::IdlAccount(acc) => acc.is_mut,
            IdlAccountItem::IdlAccounts(accs) => mutable_account_exists(&accs.accounts),
        })
```

**Review by seanyoung — APPROVED** [review]

File: src/bin/idl/mod.rs (hunk 2 above)
Comment (LucasSte) [reply src/bin/idl/mod.rs:332]: > Why do you not like iterators?

I am not used to them. I am used to writing `for(int I=0; i<n; i++)` in C, so I often write for-loops in Rust too.

**Pull request merged**